nuxeo-common/src/main/java/org/nuxeo/common/utils/TextTemplate.java

```diff
@@ -426,6 +426,15 @@ public void write(char[] cbuf) throws IOException {
                 if (filename.endsWith(ext)) {
                     processAsFreemarker = true;
                     out = new File(out.getCanonicalPath().replaceAll("\\.*" + Pattern.quote(ext) + "$", ""));
+                    if (filename.equals("." + ext.toLowerCase())) {
+                        throw new IOException("Template file : '"
+                                + in.getAbsolutePath()
+                                + "' cannot be parsed. Launcher had to stop, please fix this problem."
+                                + System.getProperty("line.separator")
+                                + "It may be caused by its output name '"
+                                + out.getAbsolutePath()
+                                + "'");
+                    }
```

Comment:
```
             for (String ext : freemarkerExtensions) {
                 if (filename.endsWith(ext)) {
                     processAsFreemarker = true;
-                    out = new File(out.getCanonicalPath().replaceAll("\\.*" + Pattern.quote(ext) + "$", ""));
                     if (filename.equals("." + ext.toLowerCase())) {
-                        throw new IOException("Template file : '"
-                                + in.getAbsolutePath()
-                                + "' cannot be parsed. Launcher had to stop, please fix this problem."
-                                + System.getProperty("line.separator")
-                                + "It may be caused by its output name '"
-                                + out.getAbsolutePath()
-                                + "'");
+                        throw new IOException(String.format("Extension only as a filename is not allowed: %s", in, filename));
                     }
+                    out = new File(out.getCanonicalPath().replaceAll("\\.*" + Pattern.quote(ext) + "$", ""));
                     break;
                 }
             }

```
You can be more accurate in the error message:

     Extension only as a filename is not allowed: /home/julien/workspace/nuxeo/nuxeo-common/bin/test/test-nxftl/.nxftl

instead of:

    Template file : '/home/julien/workspace/nuxeo/nuxeo-common/bin/test/test-nxftl/.nxftl' cannot be parsed. Launcher had to stop, please fix this problem.
    It may be caused by its output name '/tmp'